src/main/java/spoon/support/reflect/reference/CtReferenceImpl.java

```diff
@@ -24,13 +24,17 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Objects;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import static spoon.reflect.path.CtRole.NAME;
 
 public abstract class CtReferenceImpl extends CtElementImpl implements CtReference, Serializable {
 
 	private static final long serialVersionUID = 1L;
+	private static final Pattern IS_ARRAY_OR_INSTANCE = Pattern.compile("\\[\\]|@");
+	private static final Pattern IS_INNER_OR_GENERIC = Pattern.compile("\\.|<|>");
```

Comment:
These two patterns are not used to identify arrays or generics, so I think the names are a bit misleading. Could you rename them appropriately? E.g. `IS_INNER_OR_GENERIC` would be more fittingly named `NESTED_OR_GENERIC_SPLITTER` or something like that.